From b7f6b3e504401ff6f49abd709d33b47293621221 Mon Sep 17 00:00:00 2001
From: Eddie Cai <eddie.cai.linux@gmail.com>
Date: Thu, 21 May 2020 09:19:12 +0800
Subject: [PATCH 21/28] clients: desktop-shell: use second clock format by
 default

Signed-off-by: Eddie Cai <eddie.cai.linux@gmail.com>
---
 clients/desktop-shell.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/clients/desktop-shell.c b/clients/desktop-shell.c
index bde5dc8..e9ec008 100644
--- a/clients/desktop-shell.c
+++ b/clients/desktop-shell.c
@@ -51,7 +51,7 @@
 
 #include "weston-desktop-shell-client-protocol.h"
 
-#define DEFAULT_CLOCK_FORMAT CLOCK_FORMAT_MINUTES
+#define DEFAULT_CLOCK_FORMAT CLOCK_FORMAT_SECONDS
 #define DEFAULT_SPACING 10
 
 extern char **environ; /* defined by libc */
-- 
2.20.1

