From 58f790d0d111f8b69332bb1119f53548ff6491bb Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Tue, 14 Jul 2020 15:52:00 +0800
Subject: [PATCH] pixman-renderer: Support ABGR and XBGR dmabuf formats

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 libweston/pixman-renderer.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/libweston/pixman-renderer.c b/libweston/pixman-renderer.c
index dd1a6b9..9ae2b51 100644
--- a/libweston/pixman-renderer.c
+++ b/libweston/pixman-renderer.c
@@ -672,6 +672,12 @@ pixman_renderer_attach_dmabuf(struct weston_surface *es,
 	case DRM_FORMAT_XRGB8888:
 		pixman_format = PIXMAN_x8r8g8b8;
 		break;
+	case DRM_FORMAT_ABGR8888:
+		pixman_format = PIXMAN_a8b8g8r8;
+		break;
+	case DRM_FORMAT_XBGR8888:
+		pixman_format = PIXMAN_x8b8g8r8;
+		break;
 	case DRM_FORMAT_YUYV:
 		pixman_format = PIXMAN_yuy2;
 		break;
-- 
2.20.1

